### PR TITLE
Add coverage to CI

### PR DESCRIPTION
**Pull request by mimi1vx**

### CODE REVIEW

**Review by Martchus — COMMENTED** [review]
I'm wondering whether we really need *that* much fixtures. From my experience with openQA I'd say the risk of adding too many fixtures is slowing down tests unnecessarily and having sometimes a hard time understanding the test setup.

**Comment by mimi1vx** [comment]
@Martchus fixtures are based on real data, and pretty simple :D 
*input* - what bot gets from various api
*output*  - excepted results

no need any excessive tests setup/etc because bot itself is simple stateless app which acts basically like filter -> get data from some sources , modify and then forward to target

**Comment by baierjan** [comment]
> I'm wondering whether we really need _that_ much fixtures.

Using real data and known valid results would be much faster (and maybe also safer) solution than trying to create a perfect handpicked set of minimal fixtures. Not mentioning the complexity if you want to cover every possible combination of states. On the other, I am not saying it wouldn't be nice to have some "minimal working example".

I think I can live with the proposed changes.

**Comment by Martchus** [comment]
The idea wasn't to craft some artificial fixtures but to strip the production data down to what's really needed. Likely many arrays in that JSON document could be way shorter without losing test coverage.

**Review by okurz — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,5 +11,13 @@ jobs:
...
+      - name: Setup enviroment
+        run: pip install -r requirements-dev.txt 
+      - name: Run tests + cov report
+        run: python3 -m pytest -v --cov=./openqabot --cov-report=xml --cov-report=term
+      - name: Check style
+        run: black --check ./
+      - name: Upload coverage to CodeCov
+        uses: codecov/codecov-action@v2
+        with:
+          files: coverage.xml
```
[the remark this answers is not in the document]
Comment (okurz) [reply .github/workflows/ci.yml:23]: But the thing is we should have a single command that runs tests + black. I mentioned three alternatives. You could choose one or add a fourth alternative :)

[the remark this answers is not in the document]
Comment (Martchus) [reply .github/workflows/ci.yml:23]: I also prefer having a single command to run everything testing related (including style checks) *and* still be able to run `pytest` manually.

[the remark this answers is not in the document]
Comment (mimi1vx) [reply .github/workflows/ci.yml:23]: > But the thing is we should have a single command that runs tests + black. I mentioned three alternatives. You could choose one or add a fourth alternative :)

why with one command ? tests and style are two different things. If is so important run it with one command .. we can add `pytest-black` plugin to requirements and run it as one ...

[the remark this answers is not in the document]
Comment (perlpunk) [reply .github/workflows/ci.yml:23]: Having a Makefile with `test*` targets is a good language agnostic thing to get a quick overview what tests are available. Plus command line tab completion.
I don't want to remember which black, flake, doctests or whatever a certain project requires.

[the remark this answers is not in the document]
Comment (kalikiana) [reply .github/workflows/ci.yml:23]: I second Tina's point.

File: Makefile [hunk 2]
```diff
@@ -4,3 +4,11 @@ all:
...
 test:
 	python3 -m pytest
+
+.PHONY: checkstyle
+checkstyle:
+	black --check ./
+
+.PHONY: tidy
+tidy:
+	black ./
```
Comment (okurz) [inline Makefile:14]: I suggest to use the same style as in
https://github.com/os-autoinst/openQA/blob/master/Makefile#L139= , e.g. include style checks as part of test

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (okurz) [reply .github/workflows/ci.yml:23]: And now please call the Makefile from here to not duplicate

**Review by okurz — CHANGES_REQUESTED** [review]

**Review by Martchus — APPROVED** [review]

**Review by okurz — APPROVED** [review]

**Pull request merged**